### DISCUSSION TITLE
MacOS MDM - Silent Service Install?

### DISCUSSION THREAD

**Discussion by Jooga02** [rustdesk/rustdesk#5235] [Q&A]
I have zero issues getting the RustDesk client installed via Jamf Pro, this includes setting our custom settings, like our on prem server, key, id, static pass, ect.... What i'm having issues with is actually getting the service to install silently.  Does anyone know how to do this or could point me toward a guide?  I've wrestled with this for a day or so now and i'm not having much luck.  We could operate as is but we'd like the service to go ahead and install siliently.

**Comment by dinger1986** [comment]
The nightly builds have a new flag --install-service admittedly I haven't tested yet but it's here https://github.com/rustdesk/rustdesk/blob/master/src/core_main.rs line 217.

Can you share your script you are using?

**Comment by nico-hl** [comment]
Oh yes, please share your script :) 
And do anyone know where I can find the config file (change the relay server) on macOS ? I know I could change it in the app but when I have the config I can change it via command line

**Comment by majluk** [comment]
Does anyone was able to install and start service from CLI?
**sudo /Applications/RustDesk.app/Contents/MacOS/RustDesk --install-service** - Seems to not working.

**Comment by JDL1987** [comment]
> I have zero issues getting the RustDesk client installed via Jamf Pro, this includes setting our custom settings, like our on prem server, key, id, static pass, ect.... What i'm having issues with is actually getting the service to install silently. Does anyone know how to do this or could point me toward a guide? I've wrestled with this for a day or so now and i'm not having much luck. We could operate as is but we'd like the service to go ahead and install siliently.

May I ask how did you were able to install it with jamf?
I got stuck on creating a Privacy Preference Policy Control withing the configuration profiles. It always fails stating " the key "CodeRequirement" has an invalid value.
![coderequirementrust](https://github.com/user-attachments/assets/ba42351d-f540-4a0e-ba7c-cea38e0b5876)
